internal/service/streamprocessor/resource.go

```diff
@@ -0,0 +1,254 @@
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	pendingStates := []string{CreatedState}
+	desiredState := []string{}
+	if plan.State.Equal(state.State) || !updatedStateOnly(&plan, &state) {
+		resp.Diagnostics.AddError("updating a Stream Processor is not supported", "")
+		return
+	}
+	switch plan.State.ValueString() {
+	case StartedState:
+		desiredState = append(desiredState, StartedState)
+		pendingStates = append(pendingStates, StoppedState)
+		_, _, err := connV2.StreamsApi.StartStreamProcessorWithParams(ctx,
+			&admin.StartStreamProcessorApiParams{
+				GroupId:       plan.ProjectID.ValueString(),
+				TenantName:    plan.InstanceName.ValueString(),
+				ProcessorName: plan.ProcessorName.ValueString(),
+			},
+		).Execute()
+		if err != nil {
+			resp.Diagnostics.AddError("Error starting stream processor", err.Error())
+		}
+	case StoppedState:
+		desiredState = append(desiredState, StoppedState)
+		pendingStates = append(pendingStates, StartedState)
+		_, _, err := connV2.StreamsApi.StopStreamProcessorWithParams(ctx,
+			&admin.StopStreamProcessorApiParams{
+				GroupId:       plan.ProjectID.ValueString(),
+				TenantName:    plan.InstanceName.ValueString(),
+				ProcessorName: plan.ProcessorName.ValueString(),
+			},
+		).Execute()
+		if err != nil {
+			resp.Diagnostics.AddError("Error stopping stream processor", err.Error())
+		}
+	default:
+		resp.Diagnostics.AddError("transitions to states other than STARTED or STOPPED are not supported", "")
+		return
+	}
+
+	projectID := plan.ProjectID.ValueString()
+	instanceName := plan.InstanceName.ValueString()
+	requestParams := &admin.GetStreamProcessorApiParams{
+		GroupId:       projectID,
+		TenantName:    instanceName,
+		ProcessorName: plan.ProcessorName.ValueString(),
+	}
+
+	streamProcessorResp, err := WaitStateTransition(ctx, requestParams, connV2.StreamsApi, pendingStates, desiredState)
+	if err != nil {
+		resp.Diagnostics.AddError("Error changing state of stream processor", err.Error())
+	}
+
+	newStreamProcessorModel, diags := NewStreamProcessorWithStats(ctx, projectID, instanceName, streamProcessorResp, plan.Options)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newStreamProcessorModel)...)
+}
+
+func (r *streamProcessorRS) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var streamProcessorState *TFStreamProcessorRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &streamProcessorState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	if _, err := connV2.StreamsApi.DeleteStreamProcessor(ctx, streamProcessorState.ProjectID.ValueString(), streamProcessorState.InstanceName.ValueString(), streamProcessorState.ProcessorName.ValueString()).Execute(); err != nil {
+		resp.Diagnostics.AddError("error deleting resource", err.Error())
```

Comment:
missing 404 case when it has been deleted outside TF

Reply:
But this should be in the Read right? In the read the 404 case is handled. I noticed that an error was not being added in case of 404 to the Diagnostics. I will add that.

Reply:
Sorry, Yes handling it in the read should be enough. I don't think should include that diagnostic message. We don't do that in other resources